fs/tarfs.py

```diff
@@ -280,9 +281,19 @@ def _directory(self):
         if self._directory_cache is None:
             _decode = self._decode
             _directory = ((_decode(info.name).strip("/"), info) for info in self._tar)
-            self._directory_cache = OrderedDict(
-                (name, info) for name, info in _directory if normpath(name)
-            )
+
+            def _list_tar():
+                for name, info in _directory:
+                    try:
+                        _name = normpath(name)
+                    except IllegalBackReference:
+                        # Back references outside root, must be up to no good.
+                        pass
```

Comment:
Rather than _always_ silently ignoring back-reference errors, I wonder if it'd be worth ignoring them by default, with an FS init-option to re-throw the back-reference errors?

Reply:
I pondered that, but if we raise an error at this point then a tar with a single badly formatted entry would prevent the tar from being opened. At least this way you could access any other valid entries via the TarFS.

I'm assuming that an invalid path here would be a fairly exotic thing, and there wouldn't be a great utility in reporting it back to the user. As far as I can tell, the tarfile module also silently ignores these errors, so that swayed me.

I'll wait until somebody asks for it or I can think of an argument against ignoring bad entries...